### ISSUE TITLE
extended class is being overwritten

### ISSUE THREAD

**Issue by j3ddesign** [typestack/typedi#56]
First off, I love this project, it's been incredible.
I'm having a slight issue and was wondering If I could get some insight as it's probably a me issue. 
Below is a small snippet of what I have.

```
// Whitelist.ts
@Service()
export class Whitelist extends Rule {
  entries: Set<string>;

  setup(entries: WhitelistEntryFragment[]) {
    this.entries = new Set(entries.map(entry => entry.value));
  }
}
```
```
// rule.ts
@Service()
export class Rule {
  id: number;
  name: string;
  enabled: boolean;
  silent: boolean;
  settings: any;
  responses: Responses[];

  @Inject('logger')
  logger: LoggerInterface;

  static parseRule(rule: RuleFragment): RuleInterface {
    return Object.assign({}, rule, {
      responses: rule.responses.map<Responses>(response => response.value)
    });
  }
}
```

However when I run the following
```
const whitelist = Container.get(Whitelist);
console.log(whitelist);
whitelist.setup(
  data.channel.whitelist
);
```

The log returns only 
```
Rule {
  logger:
   ConsoleLogger {
     loggingLevel: 'trace',
     trace: [Function],
     debug: [Function],
     info: [Function],
     warn: [Function],
     error: [Function],
     fatal: [Function] } }
[13:49] error: TypeError: whitelist.setup is not a function
```

Any help would be appreciated here. I'm guessing I'm missing some really basic concept here but I've been lost for about an hour now 👎

**Comment by j3ddesign** [comment]
Welp. I resolved it by removing `@Service` from `Rule`.
I'm going to leave the issue open in case it isn't the intended behaviour. Thanks again for an incredible tool 👍

**Comment by pleerock** [comment]
It was a bug and is fixed in `0.7.2`. Thanks for the report.
